### PR TITLE
docs: use doxygen-awesome theme

### PR DESCRIPTION
**Pull request by paroj**
https://pullrequest.opencv.org/buildbot/export/pr/21065/docs/

https://jothepro.github.io/doxygen-awesome-css/

<cut/>

### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or other license that is incompatible with OpenCV
- [x] The PR is proposed to proper branch
- [ ] There is reference to original bug report and related work
- [ ] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [x] The feature is well documented and sample code can be built with the project CMake

```
force_builders_only=docs
```

### CODE REVIEW

File: doc/Doxyfile.in [hunk 1]
```diff
@@ -134,7 +134,7 @@ HTML_FILE_EXTENSION    = .html
 HTML_HEADER            = @CMAKE_CURRENT_SOURCE_DIR@/header.html
 HTML_FOOTER            = @CMAKE_CURRENT_SOURCE_DIR@/footer.html
 HTML_STYLESHEET        =
-HTML_EXTRA_STYLESHEET  = @CMAKE_CURRENT_SOURCE_DIR@/stylesheet.css
+HTML_EXTRA_STYLESHEET  = @CMAKE_CURRENT_SOURCE_DIR@/doxygen-awesome.css
```
Comment (alalek) [inline doc/Doxyfile.in:137]: BTW, no miracle happened, [this page](http://pullrequest.opencv.org/buildbot/export/pr/21065/docs/d4/d1c/namespacecv_1_1gapi.html) is still look broken on FHD resolution (1920x1080).

**Comment by mshabunin** [comment]
Overall looks cool. I have found several minor issues:
* Python functions are not visible, perhaps font color of `<table class="python_language">` should be changed to something brighter:
  image
* Notes are very bright compared to overall level, perhaps they could be made dimmer, it is `<dl class="section note">`:
  image
* Some images have transparent background - I'm not sure how to deal with them:
  image
* Perhaps link to the theme could be added in the footer.

File: doc/Doxyfile.in (hunk 1 above)
Comment (mshabunin) [reply doc/Doxyfile.in:137]: Perhaps complex types like `std::enable_if< cv::detail::accepted_infer_types< Input >::value, cv::GInferListOutputs >::type`  should be replaced with shorter aliases using cosmetic typedefs.

**Comment by mshabunin** [comment]
iframes in JS tutorials have own style (`<link href="js_example_style.css" rel="stylesheet" type="text/css">`):
image

**Comment by paroj** [comment]
> Overall looks cool. I have found several minor issues:
> 
>     * Python functions are not visible, perhaps font color of `<table class="python_language">` should be changed to something brighter:
>       image
> 
>     * Notes are very bright compared to overall level, perhaps they could be made dimmer, it is `<dl class="section note">`:
>       image
> 
>     * Some images have transparent background - I'm not sure how to deal with them:
>       image
> 
>     * Perhaps link to the theme could be added in the footer.

one could solve these issues by forcing the "bright" colour scheme https://github.com/jothepro/doxygen-awesome-css/issues/54
currently the theme will follow your browser preference

**Comment by catree** [comment]
> Overall looks cool. I have found several minor issues:
> 
>     * Python functions are not visible, perhaps font color of `<table class="python_language">` should be changed to something brighter:
>       image
> 
>     * Notes are very bright compared to overall level, perhaps they could be made dimmer, it is `<dl class="section note">`:
>       image
> 
>     * Some images have transparent background - I'm not sure how to deal with them:
>       image
> 
>     * Perhaps link to the theme could be added in the footer.

Are there many of images with transparent background?
If most of the PNG images with transparent background come from my contributions, I should have the originals somewhere and be able to add a white background in Gimp.

**Comment by mshabunin** [comment]
> Are there many of images with transparent background?

I don't know, probably not many. It should be possible to automate transparency removal using ImageMagick: https://stackoverflow.com/questions/2322750/replace-transparency-in-png-images-with-white-background

**Comment by alalek** [comment]
Links on pages with issues would be really useful...

May be we could force `background-color: rgba(255, 255, 255, 255);` for all images:

```.diff
 .contents img {
   max-width: 100%;
+  background-color: rgba(255,255,255,255);
 }
```

**Comment by catree** [comment]
> Quickly, I tried downloading the PNG image, adding a white layer in Gimp in background, but was not successful.
> Maybe there is an issue on my side and it should be possible to use Gimp to add a white background?

I am dumb, I was using the screenshot image instead of the original image.

---

For historical purposes, this old thread: https://github.com/LorenaGdL/opencv/issues/1 about documentation.

**Comment by paroj** [comment]
I just dropped the dark-color scheme from the theme, so the image-background issues can be handled independent of the general theme switch

**Review by mshabunin — APPROVED** [review]
Looks _awesome_! :smiley:  Thank you!

**Pull request merged**